tests/integration/multicluster/gateway.go

```diff
@@ -0,0 +1,81 @@
+// Copyright Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package multicluster
+
+import (
+	"context"
+	"fmt"
+	"path/filepath"
+	"testing"
+	"time"
+
+	"istio.io/istio/pkg/test/env"
+	"istio.io/istio/pkg/test/framework"
+	"istio.io/istio/pkg/test/framework/components/istio"
+	"istio.io/istio/pkg/test/framework/components/istioctl"
+	"istio.io/istio/pkg/test/framework/features"
+	"istio.io/istio/pkg/test/framework/label"
+	"istio.io/istio/pkg/test/scopes"
+	"istio.io/istio/pkg/test/util/retry"
+
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// GatewayTest tests that istio ingress gateway can be started successfully in remote cluster
+func GatewayTest(t *testing.T, feature features.Feature) {
+	framework.NewTest(t).
+		Label(label.Multicluster).
+		Features(feature).
+		Run(func(ctx framework.TestContext) {
+			ctx.NewSubTest("gateway").
+				Run(func(ctx framework.TestContext) {
+
+					cfg, err := istio.DefaultConfig(ctx)
+					if err != nil {
+						t.Fatalf("error ")
+					}
+					clusters := ctx.Environment().Clusters()
+					args := []string{"install", "-f", filepath.Join(env.IstioSrc,
+						"tests/integration/multicluster/testdata/gateway.yaml"), "--manifests",
+						filepath.Join(env.IstioSrc, "manifests")}
+					for _, cluster := range clusters {
+						_, err = cluster.AdmissionregistrationV1().ValidatingWebhookConfigurations().Get(context.TODO(),
+							"istiod-istio-system", metav1.GetOptions{})
+						if err == nil {
```

Comment:
Why do you not fail if `err != nil`?

Reply:
This is used to find the correct cluster that we want to check gateway.  if `err != nil` we will go on finding the correct cluster.